lib/google/youtube.js

```diff
@@ -1,18 +1,35 @@
-// const { DateTime } = require('luxon')
 const { authenticate } = require('./authenticate')
 const { google } = require('googleapis')
 
-const getStreams = async ({ max = 5, status = 'upcoming' } = {}) => {
+const getStreamsData = async ({ max = 5, status = 'upcoming', pageToken = '' } = {}) => {
   const auth = authenticate('youtube')
   const youtube = google.youtube({ version: 'v3', auth })
 
   const streams = await youtube.liveBroadcasts.list({
     broadcastStatus: status,
     maxResults: max,
-    part: 'id,snippet,contentDetails,status'
+    part: 'id,snippet,contentDetails,status',
+    pageToken
   })
 
-  return streams.data.items || []
+  return streams.data || {}
 }
 
-module.exports = { getStreams }
+const getStreams = async ({ max = 5, status = 'upcoming', pageToken = '' } = {}) => {
+  const streamsData = await getStreamsData({ max, status, pageToken })
+
+  return streamsData.items || []
+}
+
+const getLastStream = async ({ max = 5, status = 'upcoming', pageToken = '' } = {}) => {
+  const streams = await getStreamsData({ max, status, pageToken })
+
+  if (streams.nextPageToken) {
+    await getLastStream({ max, status, pageToken: streams?.nextPageToken })
```

Comment:
the returning of this call if never used, so you will never really get the last scheduled stream, but only the first page.
you probably want:

```suggestion
    const nextPageStreams = await getLastStream({ max, status, pageToken: streams?.nextPageToken })
    return nextPageStreams.items?.at(-1) || null
```

Notes to consider:

- make sure you get results from `nextPageStreams` or do not return from the block
- use a smaller limit (e.g. `max: 1`) when testing to get more pages

Reply:
I changed this part, please lmk if it is correct now. 
Changed the max to 1 and it is returning the last brownbag scheduled. Just it took more time with max = 1.